include/exadg/incompressible_navier_stokes/postprocessor/pointwise_output_generator.cpp

```diff
@@ -0,0 +1,115 @@
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <exadg/incompressible_navier_stokes/postprocessor/pointwise_output_generator.h>
+#include <exadg/utilities/print_functions.h>
+
+namespace ExaDG
+{
+namespace IncNS
+{
+template<int dim>
+PointwiseOutputData<dim>::PointwiseOutputData() : write_velocity(false), write_pressure(false)
+{
+}
+
+template<int dim>
+void
+PointwiseOutputData<dim>::print(dealii::ConditionalOStream & pcout) const
+{
+  PointwiseOutputDataBase<dim>::print(pcout);
+
+  if(this->time_control_data.is_active && this->evaluation_points.size() > 0)
+  {
+    print_parameter(pcout, "Write velocity", write_velocity);
+    print_parameter(pcout, "Write pressure", write_pressure);
+  }
+}
+
+template struct PointwiseOutputData<2>;
+template struct PointwiseOutputData<3>;
+
+template<int dim, typename Number>
+PointwiseOutputGenerator<dim, Number>::PointwiseOutputGenerator(MPI_Comm const & comm)
+  : PointwiseOutputGeneratorBase<dim, Number>(comm)
+{
+}
+
+template<int dim, typename Number>
+void
+PointwiseOutputGenerator<dim, Number>::setup(
+  dealii::DoFHandler<dim> const &  dof_handler_velocity_in,
+  dealii::DoFHandler<dim> const &  dof_handler_pressure_in,
+  dealii::Mapping<dim> const &     mapping_in,
+  PointwiseOutputData<dim> const & pointwise_output_data_in)
+{
+  this->setup_base(dof_handler_pressure_in.get_triangulation(),
+                   mapping_in,
+                   pointwise_output_data_in);
+
+  dof_handler_velocity = &dof_handler_velocity_in;
+  dof_handler_pressure = &dof_handler_pressure_in;
+
+  pointwise_output_data = pointwise_output_data_in;
+
+  if(pointwise_output_data.time_control_data.is_active and
+     pointwise_output_data.evaluation_points.size() > 0)
+  {
+    if(pointwise_output_data.write_velocity)
+      this->add_quantity("Velocity", dim);
+    if(pointwise_output_data.write_pressure)
+      this->add_quantity("Pressure", 1);
+  }
+}
+
+template<int dim, typename Number>
+void
+PointwiseOutputGenerator<dim, Number>::evaluate(VectorType const & velocity,
+                                                VectorType const & pressure,
+                                                double const       time,
+                                                bool const         unsteady)
+{
+  this->do_evaluate(
+    [&]() {
+      if(pointwise_output_data.write_velocity)
+      {
+        auto const values =
+          this->template compute_point_values<dim>(velocity, *dof_handler_velocity);
+        this->write_quantity("Velocity", values, 0);
```

Comment:
What is the meaning of 0 here? Does the number here have to be consisting with another number written somewhere else in the code? There is no number for the "pressure-call" below.

In similar cases, I often add a comment to the function call for clarity/documentation.

Reply:
It is the first selected component. For the overload of `write_quantity` for scalar quantities, a first selected component is not considered.

I will add a comment.